### ISSUE TITLE
What are the limits for custom event attributes?

### ISSUE THREAD

**Issue by mbrochh** [aws/aws-amplify#822]
I found this FAQ that mentions some pretty scary limits: https://aws.amazon.com/pinpoint/faqs/

> You can have up to 1,500 unique custom event types per app and up to 40 attributes and metrics per custom event.

> Amazon Pinpoint provides 40 custom attributes. You can use these to customize your segmentation for your unique use cases. Each custom attribute can store up to 50 values.

I was planning to structure my events like this:

The event will be called "tap" or "view" or "swipe" or "scroll" - basically, a simple user action, there are only so many actions a user can do, so I will not get anywhere close to 1500 events.

Then I would use attributes like this: viewName: "ShopGridView", componentName: "Product", objectId: 123

I would not need more than 40 custom attributes, I think, but the thing that I don't understand is the 50 values per attribute. If wanted to store the objectIds or my userIDs in the attributes but I have thousands of products and thousands of users, does it mean, I can't store that kind of information in my events??

Or in simpler terms, lets say I just want to track what views my users navigate to, just like Google Analytics tracks pageviews, but since I have thousands of products, I basically have thousands of views, is there no way that I can track exactly which view a user clicked into? If I can just track "user visited a product detail view", that's not really good enough information, is it?

Would I have to store my objectIds as metrics, maybe? Would those not be limited?

**Comment by powerful23** [comment]
@mbrochh I guess the custom attributes in that faqs means the user attributes for the endpoint. https://aws.github.io/aws-amplify/media/analytics_guide#update-user-attributes

Confirming with the Pinpoint team.

**Comment by powerful23** [comment]
@mbrochh Confirmed. That limit is just for the user attributes of the pinpoint endpoint.
